src/log.cpp

```diff
@@ -179,7 +181,11 @@ void set_log_to_file()
 	if(is_log_dir_writable_.value_or(false)) {
 		// get the log file stream and assign cerr+cout to it
 		output_file_path_ = filesystem::get_logs_dir()+"/"+unique_log_filename();
-		output_file_.reset(filesystem::ostream_file(output_file_path_).release());
+		static std::unique_ptr<std::ostream> logfile { filesystem::ostream_file(output_file_path_) };
```

Comment:
Could this cause it to be deleted before the last log statement when exiting? I'm worried that it might add a Heisenbug that only triggers when trying to debug a different Heisenbug.

Reply:
Static local variables are destructed at the same time as other global variables, that is already after main() is exited. Thus this can cause trouble only if you're logging from a destructor of global variable which is already an UB because it could happen after logger is destructed.